src/arch/svm/SpokeUtils.ts

```diff
@@ -67,17 +69,72 @@ export function getDepositIdAtBlock(_contract: unknown, _blockTag: number): Prom
   throw new Error("getDepositIdAtBlock: not implemented");
 }
 
-export function findDepositBlock(
-  _spokePool: unknown,
+/**
+ * Finds deposit events within a 2-day window ending at the specified timestamp.
+ *
+ * @remarks
+ * This implementation uses a time-limited search approach because Solana PDA state has
+ * limitations that prevent directly referencing old deposit IDs. Unlike EVM chains where
+ * we might use binary search across the entire chain history, in Solana we must query within
+ * a constrained slot range.
+ *
+ * The search window is calculated by:
+ * 1. Finding the slot at the specified timestamp (or current time if no timestamp is provided)
+ * 2. Looking back 2 days worth of slots from that point
+ *
+ * We use a 2-day window because:
+ * 1. Most valid deposits that need to be processed will be recent
+ * 2. This covers multiple bundle submission periods
+ * 3. It balances performance with practical deposit age
+ *
+ * @important
+ * This function may return `undefined` for valid deposit IDs that are older than the search
+ * window (approximately 2 days before the specified timestamp). This is an acceptable limitation
+ * as deposits this old are typically not relevant to current operations.
+ *
+ * @param eventClient - SvmCpiEventsClient instance
+ * @param depositId - The deposit ID to search for
+ * @param timestamp - The timestamp to search up to (defaults to current time). The search will look
+ *                   for deposits between (timestamp - 2 days) and timestamp. Time must be in seconds.
+ * @returns The deposit if found within the time window, undefined otherwise
+ */
+export async function findDeposit(
+  eventClient: SvmCpiEventsClient,
   depositId: BigNumber,
-  _lowBlock: number,
-  _highBlock?: number
-): Promise<number | undefined> {
+  timestamp: number = getCurrentTime()
+): Promise<DepositWithBlock | undefined> {
   // We can only perform this search when we have a safe deposit ID.
   if (isUnsafeDepositId(depositId)) {
     throw new Error(`Cannot binary search for depositId ${depositId}`);
   }
-  throw new Error("findDepositBlock: not implemented");
+  const [currentTimeInMs, timestampInMs] = [getCurrentTime() * 1000, timestamp * 1000];
+  const slotDurationInMs = 300;
+
+  // We know we're likely to not be searching for a deposit id that is
+  // greater than two days older than the provided timestamp (or current time if no timestamp is provided)
+  // therefore we can use the current block and timestamp to calculate the slot at the timestamp and then
+  // calculate two days ago from that slot.
+  const provider = eventClient.getRpc();
+  const currentSlot = await provider.getSlot({ commitment: "confirmed" }).send();
+  const endSlot = currentSlot - BigInt(Math.round((currentTimeInMs - timestampInMs) / slotDurationInMs));
+  const startSlot = endSlot - BigInt(Math.round((2 * 24 * 60 * 60 * 1000) / slotDurationInMs));
```

Comment:
Quick question: are we sure the 2-day value won’t ever change? If not it might be worth making it a function param with a default of 2.